### ISSUE TITLE
'Message Details' of a received media lists your own phone number as a sender

### ISSUE THREAD

**Issue by 2-4601** [signalapp/Signal-Android#2874]
1. Open an existing conversation (regular or group, doesn't matter)
2. Find a message with media that **someone has sent to you**
3. Tap and hold the message for 'Message Details'
4. On 'Message Details' your own phone number is listed as a sender below the text 'From:'

Expected behavior is that your own number should not be there (just like it is not with plain text messages).

Affects all supported media: picture, audio and video (ok, ok, I failed at testing with video because recording a <280kb video turned out to be too cumbersome – even with the shittiest resolution a one second video was twice as large as TextSecure's size limit :)

Device info: Nexus 4, stock Android 5.0.1, TextSecure 2.9.3
(Also tested with another device that runs on 4.0.4 and has 2.9.3)

An example of an image message:
![textsecure message details sender phone](https://cloud.githubusercontent.com/assets/174176/6983141/167f87e0-da23-11e4-9d3d-335d8a37d6b1.png)

**Comment by rhodey** [comment]
confirmed on 2.8.0, happens with push & MMS media.

**Comment by rhodey** [comment]
the issue seems to be that [MmsAddressDatabase.getRecipientsForId()](https://github.com/WhisperSystems/TextSecure/blob/master/src/org/thoughtcrime/securesms/database/MmsAddressDatabase.java#L137) returns all addresses associated with a given message ID, this means both sender and receiver, but [MessageDetailsActivity](https://github.com/WhisperSystems/TextSecure/blob/master/src/org/thoughtcrime/securesms/MessageDetailsActivity.java#L260) doesn't bother to filter through the `Recipients` result based on message type (outgoing vs incoming).

**Comment by mcginty** [comment]
i'm loving these screenshots

**Comment by 2-4601** [comment]
@mcginty Thanks, I loved staging them! :D

**Comment by rhodey** [comment]
I started working on this again and discovered a more fundamental issue that results in this bug. Turns out that for sent messages the `MessageRecord` returned by `MmsDatabase.Reader` always has a recipient list of length 1 where the only entry is an unknown recipient. This is because `MmsDatabase.getMediaMmsMessageRecord()` [builds the recipient list](https://github.com/WhisperSystems/TextSecure/blob/28232fc2a5f8ac8ba11283ca9992f1052467d28b/src/org/thoughtcrime/securesms/database/MmsDatabase.java#L1037) from the `address` column which is only populated for received messages. Someone must have realized that this was the case and added what *at first* appears to be an extra db query in [MessageDetailsActivity](https://github.com/WhisperSystems/TextSecure/blob/28232fc2a5f8ac8ba11283ca9992f1052467d28b/src/org/thoughtcrime/securesms/MessageDetailsActivity.java#L260).

So, every sent media `MessageRecord` coming from our database has an invalid recipient list-- I was surprised that this worked at all and expected that if I tried to change it it many things would break, they did. After modifying `MmsDatabase.getMediaMmsMessageRecord()` to correctly build the recipient list for sent media messages all sorts of things started going wrong, most of which I didn't entirely understand, so I stopped :|

The quick fix for this would be to filter the users number from the recipient list returned by that *seeminly* extra db query in MessageDetailsActivity, but that would be something of a hack and definitely doesn't address the underlying strangeness. Removing assignment from myself because this touches some serious TS internals.
